src/core/render/index.js

```diff
@@ -12,289 +11,301 @@ import { Compiler } from './compiler';
+      const path = vm.route.path;
 
-    el.setAttribute('href', nameLink[match]);
-  }
-}
+      if (!el) {
+        return;
+      }
 
-export function renderMixin(proto) {
-  proto._renderTo = function(el, content, replace) {
-    const node = dom.getNode(el);
-    if (node) {
-      node[replace ? 'outerHTML' : 'innerHTML'] = content;
-    }
-  };
+      if (isPrimitive(vm.config.nameLink)) {
+        el.setAttribute('href', nameLink);
+      } else if (typeof nameLink === 'object') {
+        const match = Object.keys(nameLink).filter(
+          key => path.indexOf(key) > -1
+        )[0];
 
-  proto._renderSidebar = function(text) {
-    const { maxLevel, subMaxLevel, loadSidebar, hideSidebar } = this.config;
-
-    if (hideSidebar) {
-      // FIXME : better styling solution
-      document.querySelector('aside.sidebar').remove();
-      document.querySelector('button.sidebar-toggle').remove();
-      document.querySelector('section.content').style.right = 'unset';
-      document.querySelector('section.content').style.left = 'unset';
-      document.querySelector('section.content').style.position = 'relative';
-      document.querySelector('section.content').style.width = '100%';
-      return null;
+        el.setAttribute('href', nameLink[match]);
+      }
     }
 
-    this._renderTo('.sidebar-nav', this.compiler.sidebar(text, maxLevel));
-    const activeEl = getAndActive(this.router, '.sidebar-nav', true, true);
-    if (loadSidebar && activeEl) {
-      activeEl.parentNode.innerHTML +=
-        this.compiler.subSidebar(subMaxLevel) || '';
-    } else {
-      // Reset toc
-      this.compiler.subSidebar();
+    _renderTo(el, content, replace) {
+      const node = dom.getNode(el);
+      if (node) {
+        node[replace ? 'outerHTML' : 'innerHTML'] = content;
+      }
     }
 
-    // Bind event
-    this._bindEventOnRendered(activeEl);
-  };
+    _renderSidebar(text) {
+      const { maxLevel, subMaxLevel, loadSidebar, hideSidebar } = this.config;
+
+      if (hideSidebar) {
+        // FIXME : better styling solution
+        document.querySelector('aside.sidebar').remove();
+        document.querySelector('button.sidebar-toggle').remove();
+        document.querySelector('section.content').style.right = 'unset';
+        document.querySelector('section.content').style.left = 'unset';
+        document.querySelector('section.content').style.position = 'relative';
+        document.querySelector('section.content').style.width = '100%';
+        return null;
+      }
+
+      this._renderTo('.sidebar-nav', this.compiler.sidebar(text, maxLevel));
+      const activeEl = getAndActive(this.router, '.sidebar-nav', true, true);
+      if (loadSidebar && activeEl) {
+        activeEl.parentNode.innerHTML +=
+          this.compiler.subSidebar(subMaxLevel) || '';
+      } else {
+        // Reset toc
+        this.compiler.subSidebar();
+      }
+
+      // Bind event
+      this._bindEventOnRendered(activeEl);
+    }
```

Comment:
I made literally no logic changes. I only moved the code around within the file, so that all functions are now methods inside the new classes.